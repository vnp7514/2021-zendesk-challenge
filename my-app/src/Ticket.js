import { useState } from "react";

function Ticket(props){
    const [moreInfo, setMoreInfo] = useState(false);
    
    return (
        <div>
            <div>
                <p>Ticket ID: {props.ticket.id}</p>
                <p>Type: {(props.ticket.type) ? props.ticket.type : ""}</p>
                <p>Priority: {props.ticket.priority}</p>
                <p>Due at: {props.ticket.due_at}</p>
                <p>Status: {props.ticket.status}</p>
                <p>Description: {props.ticket.description}</p>
            </div>
            <div>
                {(moreInfo) ? JSON.stringify(props.ticket, null, 4) : ""}
            </div>
            <button onClick={()=>setMoreInfo(!moreInfo)}>
                {moreInfo ? "Less information" : "More information"}
            </button>
        </div>
    );
}

export default Ticket;