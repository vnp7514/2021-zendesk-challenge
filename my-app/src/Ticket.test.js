import { fireEvent, render, screen } from '@testing-library/react';
import Ticket from './Ticket';

const ticket = {
    "url": "https://zccvnp7514.zendesk.com/api/v2/tickets/1.json",
    "id": 1,
    "external_id": null,
    "via": {
        "channel": "sample_ticket",
        "source": {
            "from": {},
            "to": {},
            "rel": null
        }
    },
    "created_at": "2021-11-28T21:04:23Z",
    "updated_at": "2021-11-28T21:04:23Z",
    "type": "incident",
    "subject": "Sample ticket: Meet the ticket",
    "raw_subject": "Sample ticket: Meet the ticket",
    "description": "Hi there,\n\nI’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?\n\nThanks,\n The Customer\n\n",
    "priority": "normal",
    "status": "open",
    "recipient": null,
    "requester_id": 1902362971284,
    "submitter_id": 1267154019450,
    "assignee_id": 1267154019450,
    "organization_id": null,
    "group_id": 4411403643931,
    "collaborator_ids": [],
    "follower_ids": [],
    "email_cc_ids": [],
    "forum_topic_id": null,
    "problem_id": null,
    "has_incidents": false,
    "is_public": true,
    "due_at": null,
    "tags": [
        "sample",
        "support",
        "zendesk"
    ],
    "custom_fields": [],
    "satisfaction_rating": null,
    "sharing_agreement_ids": [],
    "fields": [],
    "followup_ids": [],
    "ticket_form_id": 1900004276604,
    "brand_id": 1900001352684,
    "allow_channelback": false,
    "allow_attachments": true
};

test('Render ticket component correctly', () => {
    const {container} = render(<Ticket ticket={ticket}/>);
    expect(container).toMatchSnapshot();
});

test('When more information button is clicked, the whole JSON is shown', () => {
    const {container, getByTestId} = render(<Ticket ticket={ticket} />);
    const button = getByTestId("ticket");
    fireEvent.click(button);
    expect(container).toMatchSnapshot();
});