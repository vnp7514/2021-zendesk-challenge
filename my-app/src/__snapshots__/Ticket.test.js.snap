// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ticket component correctly 1`] = `
<div>
  <div
    class="sc-gsDKAQ fpLdtY"
  >
    <div
      class="sc-dkPtRN iWYJRA"
    >
      <div
        class="sc-hKwDye eyUCLg"
      >
        <p>
          Ticket ID: 
          1
        </p>
        <p>
          Type: 
          incident
        </p>
        <p>
          Priority: 
          normal
        </p>
        <p>
          Status: 
          open
        </p>
      </div>
      <div
        class="sc-eCImPb iMFMIt"
      >
        <p>
          Description: 
          Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer


        </p>
      </div>
    </div>
    <div />
    <button
      class="sc-bdvvtL iNBsxJ"
      data-testid="ticket"
    >
      More information
    </button>
  </div>
</div>
`;

exports[`When more information button is clicked, the whole JSON is shown 1`] = `
<div>
  <div
    class="sc-gsDKAQ fpLdtY"
  >
    <div
      class="sc-dkPtRN iWYJRA"
    >
      <div
        class="sc-hKwDye eyUCLg"
      >
        <p>
          Ticket ID: 
          1
        </p>
        <p>
          Type: 
          incident
        </p>
        <p>
          Priority: 
          normal
        </p>
        <p>
          Status: 
          open
        </p>
      </div>
      <div
        class="sc-eCImPb iMFMIt"
      >
        <p>
          Description: 
          Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer


        </p>
      </div>
    </div>
    <div>
      {
    "url": "https://zccvnp7514.zendesk.com/api/v2/tickets/1.json",
    "id": 1,
    "external_id": null,
    "via": {
        "channel": "sample_ticket",
        "source": {
            "from": {},
            "to": {},
            "rel": null
        }
    },
    "created_at": "2021-11-28T21:04:23Z",
    "updated_at": "2021-11-28T21:04:23Z",
    "type": "incident",
    "subject": "Sample ticket: Meet the ticket",
    "raw_subject": "Sample ticket: Meet the ticket",
    "description": "Hi there,\\n\\nI’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?\\n\\nThanks,\\n The Customer\\n\\n",
    "priority": "normal",
    "status": "open",
    "recipient": null,
    "requester_id": 1902362971284,
    "submitter_id": 1267154019450,
    "assignee_id": 1267154019450,
    "organization_id": null,
    "group_id": 4411403643931,
    "collaborator_ids": [],
    "follower_ids": [],
    "email_cc_ids": [],
    "forum_topic_id": null,
    "problem_id": null,
    "has_incidents": false,
    "is_public": true,
    "due_at": null,
    "tags": [
        "sample",
        "support",
        "zendesk"
    ],
    "custom_fields": [],
    "satisfaction_rating": null,
    "sharing_agreement_ids": [],
    "fields": [],
    "followup_ids": [],
    "ticket_form_id": 1900004276604,
    "brand_id": 1900001352684,
    "allow_channelback": false,
    "allow_attachments": true
}
    </div>
    <button
      class="sc-bdvvtL iNBsxJ"
      data-testid="ticket"
    >
      Less information
    </button>
  </div>
</div>
`;
